AirCasting/Services/RemoveOldMeasurementsService.swift

```diff
@@ -0,0 +1,53 @@
+// Created by Lunar on 26/05/2021.
+//
+
+import Foundation
+import CoreData
+
+
+final class RemoveOldMeasurementsService {
+    
+    enum Error: Swift.Error {
+        case sessionFetchFailed
+        case removingMeasurementsSurplusFailed
+        
+    }
+    private let twentyFourHoursMeasurementCount = 24 * 60
+    
+    func removeOldestMeasurements(context: NSManagedObjectContext, uuid: SessionUUID) {
+        /// In fixed sessions we need to remove measurements older than 24h, but we treat 24 not like a date, but as a sum of measurements.
+        /// We know that we have 60 measurements per hour, so we take 1440 last measurements for each stream, and we remove older than the first of them.
+        do {
+            let session = try fetchSession(context: context, uuid: uuid)
+            removeMeasurementsSurplus(context: context, session: session)
+        } catch {
+            Log.error("Error: \(Error.removingMeasurementsSurplusFailed)")
```

Comment:
Seams like if the error would be propagated forward, the users of `removeOldestMeasurements` whould be responsible to handle it. In some case it could end with a log but in case of a log in the upper level the log could contain more reelable context. And some users would like to know that it fail and do something more with it. For example display an error to a user, react with a fallback logic or even crash if it failure could lead to larger problems.